### ISSUE TITLE
Bug: Arc Layer projection issue in 8.4.1

### ISSUE THREAD

**Issue by heshan0131** [visgl/deck.gl#5472]
#### Description
Arc is cut off when zoomed in close. This happens when upgraded to 8.4.1

![ezgif com-gif-maker (7)](https://user-images.githubusercontent.com/3605556/107740302-c1693280-6d45-11eb-82a1-8ef865530e11.gif)

UPD: The issue is in arc-layer-vertex.glsl.js
```
if (useShortestPath) {
      if (nextPos.x < 0.0) {
        currPos.x += TILE_SIZE;
        nextPos.x += TILE_SIZE;
      }
    }
```

It probably doesn’t work as expected with all projection modes. With lower zoom, we use WebMercator projection mode/cord system, with higher zoom switching to Cartesian. It looks like for cartesian the center is where the camera points and everything < 0 is cut shifted.

Should we add `project_uCoordinateSystem != COORDINATE_SYSTEM_CARTESIAN` here?

#### Repro Steps
https://codepen.io/heshan0131/pen/gOLLQzz

#### Environment
 - Framework Version: deck.gl 8.4.1

**Comment by heshan0131** [comment]
@igorDykhta

**Comment by igorDykhta** [comment]
Looking on project.glsl it looks like in this case the shift happens not for Cartesian coord system, but because of auto offset:

`project_uProjectionMode == PROJECTION_MODE_WEB_MERCATOR_AUTO_OFFSET && project_uCoordinateSystem == COORDINATE_SYSTEM_LNGLAT`

**Comment by Pessimistress** [comment]
Fixed in 8.4.5
